runtime/near-vm-runner/src/tests/cache.rs

```diff
@@ -102,19 +102,58 @@ fn test_wasmer2_artifact_output_stability() {
     //
     // Note that this test is a best-effort fish net. Some changes that should modify the hash will
     // fall through the cracks here, but hopefully it should catch most of the fish just fine.
-    let mut hasher = StableHasher::new();
-    let code = near_test_contracts::trivial_contract();
-    let config = VMConfig::test();
-    let prepared_code = prepare::prepare_contract(code, &config).unwrap();
-    let mut features = CpuFeature::set();
-    features.insert(CpuFeature::AVX);
-    let triple = "x86_64-unknown-linux-gnu".parse().unwrap();
-    let target = Target::new(triple, features);
-    let vm = Wasmer2VM::new_for_target(config, target);
-    let artifact = vm.compile_uncached(&prepared_code).unwrap();
-    let serialized = artifact.artifact().serialize().unwrap();
-    serialized.hash(&mut hasher);
-    assert_eq!(hasher.finish(), 16203733374745522118, "WASMER2_CONFIG needs version change");
+    let expected_hashes = [
+        18202109167614012273,
+        6447014787838102836,
+        11697107130663444012,
+        12595491324983981419,
+        16474677138972935760,
+        9454591335905396305,
+        1559551864276181101,
+    ];
+    let seeds = [2, 3, 5, 7, 11, 13, 17];
+    let input_hashes = [
+        13685187424711381356,
+        14874064587977634803,
+        14533057556711812527,
+        878586650292194527,
+        3964709159845744664,
+        18149828446803675487,
+        4708788108836517955,
+    ];
+    let mut got_input_hashes = Vec::with_capacity(seeds.len());
+    let mut got_hashes = Vec::with_capacity(seeds.len());
+    for seed in seeds {
+        let mut hasher = StableHasher::new();
+        let contract = near_test_contracts::arbitrary_contract(seed);
+        let mut ch = StableHasher::new();
+        contract.hash(&mut ch);
+        let config = VMConfig::test();
+        let prepared_code = prepare::prepare_contract(&contract, &config).unwrap();
```

Comment:
I wonder if we want to add

```
let hash = StableHasher::stable_hash((&contract, &prepared_code))
```

API for convenience, while we are at it? It's a bit hard to keep track of hasher and ch.

Reply:
I made it better by reordering the code and adding newlines.